### PR TITLE
[PSA Limits] Support Prefilled Amount for Withdrawal SDK

### PR DESCRIPTION
**Pull request by rebekahlow-jy**
## Approach
- If params for withdrawalFlow specifies withdrawalAmount, prefill the amount that the user is going to withdraw

```
  const xfersComponents = new Components('xfers_elements', apiToken, {
      country: country,
      customBaseDomain: currentBaseUrl,
      embed: true
    })

    const withdrawalOptions = {
      withdrawalAmount: 1000
    }

    xfersComponents.embedWithdrawalFlow(withdrawalOptions)
```

## Screenshots
GeneralWallet

### CODE REVIEW

**Comment by rebekahlow-jy** [comment]
@donbader the Withdrawal SDK will call the this API `'POST', 'user/bank_account/' + params.bankId + '/withdraw'` aka `api/v3/users_controller#withdrawal_using_new_transactional_model`. Inside that method there's this part, where `account = @xfers_api_token.account`, which API token do you want me to initialise the Withdrawal SDK (with pre-filled amount) with ah? Since we're planning to do the withdrawal from the user's `pending_withdrawal` account. 
```
      withdrawal_contract_json[:comment] = nil
      withdrawal_contract_json[:wallet_name] = account.wallet_type.name
      withdrawal_contract_json[:express] = withdrawal_contract.request_type == "FAST"
```

Do we need to create a new Xfers API Token for the user linked to the user's `pending_withdrawal` account? This will affect the Withdrawal Confirmation page as well, you can see from the GIF that it's showing as `Transfer From: Your General Wallet Account` since I was doing this demo using the user's Xfers API Token connected to their General Wallet account.

File: JavaScript/src/index.js [hunk 1]
```diff
@@ -238,9 +238,9 @@ export class Components {
 
   startTopUpFlow = () => this.element.openModal(WALLET_FLOW, { flowType: 'topup' })
 
-  embedWithdrawalFlow = () => this.element.embedView(WALLET_FLOW, { flowType: 'withdrawal' })
+  embedWithdrawalFlow = params => this.element.embedView(WALLET_FLOW, { flowType: 'withdrawal', ...params })
```
Comment (alxtz) [inline JavaScript/src/index.js]: [question] spreading params are usually bad & not readable, what's the use case for this?

Comment (rebekahlow-jy) [reply JavaScript/src/index.js]: it's for us to specify additional params when initiating the Withdrawal SDK, in this case we're taking in withdrawalAmount haha

File: JavaScript/src/components/flows/WalletFlow/actions/withdrawalActions.js [hunk 2]
```diff
@@ -81,15 +78,21 @@ export const fetchFeeBreakdown = callback => (dispatch, getState) => {
...
 
   const express = newWithdrawalRequest.withdrawalProcessingType == 'express' ? true : false
+  const PENDING_WITHDRAWAL_WALLET_ID = 9
+
+  let withdrawalParams = {
+    bankId: newWithdrawalRequest.withdrawalBankId,
+    amount: newWithdrawalRequest.withdrawalAmount,
+    express,
+    // Withdrawal via User's Pending Withdrawal Account
+    ...(params.withdrawalAmount && { wallet_id: PENDING_WITHDRAWAL_WALLET_ID })
```
Comment (rebekahlow-jy) [inline JavaScript/src/components/flows/WalletFlow/actions/withdrawalActions.js]: @donbader ~~~

**Pull request merged**